File: pset6/sentimental/caesar.py
from cs50 import get_string
import sys

while True:
    if len(sys.argv) != 2:
        print("Error: Must provide key as command line argument.")
        sys.exit(1)
    plaintext = get_string("plaintext: ")
    if plaintext:
        break

key = int(sys.argv[1])

print("ciphertext: ", end="")
for c in plaintext:
    if c.isalpha() == True:
        if c.isupper() == True:
            ascii = ord(c)
            ascii -= 65
            c = (ascii + key) % 26
            c += 65
            c = chr(c)
            print(c, end="")
        if c.islower() == True:
            ascii = ord(c)
            ascii -= 97
            c = (ascii + key) % 26
            c += 97
            c = chr(c)
            print(c, end="")
    else:
        print(c, end="")
print()

File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h> //isalpha
#include <stdio.h>
#include <stdlib.h> //atoi
#include <string.h> //strlen

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Error! Specify key in command line argument.\n");
        return 1;
    }
    else
    {
        string key = argv[1];
        int k = atoi(key);
        string plaintext = get_string("plaintext:  ");
        printf("ciphertext: ");
        for(int i=0, n = strlen(plaintext); i < n; i++)
        {
        if (isalpha(plaintext[i]))
            {
                if (isupper(plaintext[i]))
                {
                    printf("%c", (((plaintext[i] - 65) + k)%26 + 65));
                }
                if (islower(plaintext[i]))
                {
                    printf("%c", (((plaintext[i] - 97) + k)%26 + 97));
                }
            }
        else
            {
                printf("%c", plaintext[i]);
            }
        }
        printf("\n");
    }
}
